google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -1093,8 +1094,11 @@ public void testExtractJob() throws InterruptedException, TimeoutException {
     Job remoteExtractJob = bigquery.create(JobInfo.of(extractConfiguration));
     remoteExtractJob = remoteExtractJob.waitFor();
     assertNull(remoteExtractJob.getStatus().getError());
-    assertEquals(CSV_CONTENT,
-        new String(storage.readAllBytes(BUCKET, EXTRACT_FILE), StandardCharsets.UTF_8));
+
+    // Don't compare strings directly; BigQuery might reorder.
+    String gotCsvContent = new String(storage.readAllBytes(BUCKET, EXTRACT_FILE), StandardCharsets.UTF_8);
+    Truth.assertThat(gotCsvContent.split("\n")).asList().containsExactly(CSV_CONTENT.split("\n"));
```

Comment:
The test fails after I made my change but the root cause seems unrelated to the change. Tim, can you verify that export to CSV can reorder rows?

Reply:
I don't think BigQuery gives any guarantees about row order in an export job, but I can check with the backend folks.